src/synapse/api.py

```diff
@@ -48,26 +57,61 @@ class NetworkError(APIError):
 
 
 class GetNonceError(APIError):
-    """Exception raised when getting nonce via API fails."""
+    """Exception raised when getting nonce fails."""
 
 
 class GetVersionError(APIError):
-    """Exception raised when getting version via API fails."""
+    """Exception raised when getting version fails."""
 
 
 class VersionUnexpectedContentError(GetVersionError):
     """Exception raised when output of getting version is unexpected."""
 
 
-def register_user(registration_shared_secret: str, user: User) -> None:
+class GetRoomIDError(APIError):
```

Comment:
I'm not seeing where these specific exceptions are leveraged to justify not raising an `APIError` instead and simplify the code

Reply:
I don't remember if this was recommended in one previous review to be honest :-( 
@gregory-schiano @jdkandersson suggestions? Can we use APIError?

Reply:
It depends on if it is useful to distinguish between APIError and this particular error during the handling of the exception

Reply:
Since this is something that involves other contexts, I created a story (ISD-1070) so this can be addressed in a follow-up PR.